### PR TITLE
[INLONG-WEBSITE-764][DataProxy] Add Golang SDK documents

### LINKED ISSUE #764: [Feature][DataProxy] Add Golang SDK documents
### Description

Add documents for DataProxy Golang SDK

### Use case

Add documents for DataProxy Golang SDK to guide users how to use the SDK to produce messages to InLong DataProxy.

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by gunli**
### [INLONG-WEBSITE-764][DataProxy] Add Golang SDK documents

- Fixes #764

### Motivation

I have push a PR to add an INLONG DataProxy Golang SDK, and here adds the relative documents.

### Modifications

Add docs/sdk/dataproxy-sdk/go.md
Add i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change added tests and can be verified as follows:
 - Merge the PR files.
 - Verify if the newly added file can be showed on the website.

### Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)
  - If a feature is not applicable for documentation, explain why? This is a documents PR.

### CODE REVIEW

**Comment by dockerzhang** [comment]
related to https://github.com/apache/inlong/pull/8111

File: i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+## 新建实时同步任务
+在 Dashboard 或者通过命令行工具创建任务，数据源类型使用 `Auto Push` (自主推送)。
+
+## 引入 Go SDK
+需要在项目中导入SDK 的包，进行 SDK 的使用。导入方式如下：
+
+```go
+import "github.com/apache/inlong/inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/dataproxy"
+```
+
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md:17]: Too many empty lines.

File: i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
+---
+title: Go SDK
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md:2]: Go SDK
->
Golang SDK

File: i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+sidebar_position: 4
+---
+
+import {siteVariables} from '../../version';
+
+## 新建实时同步任务
+在 Dashboard 或者通过命令行工具创建任务，数据源类型使用 `Auto Push` (自主推送)。
+
+## 引入 Go SDK
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md:11]: ditto

File: i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+```go
+import "github.com/apache/inlong/inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/dataproxy"
+```
+
+
+
+
+## 数据上报流程
+导入 SDK 后，通过实例化一个[Client](https://github.com/apache/inlong/tree/master/inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/dataproxy/client.go)接口对象后，调用相关的同步（`Send()`）或 异步（`SendAsync()`）接口来完成单条数据的上报任务，SDK内部会根据StreamID将相同StreamID的数据批量上报。发送Demo可参考 [example_test.go](https://github.com/apache/inlong/tree/master/inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/dataproxy/example_test.go)。
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md:22]: SDK内部会根据StreamID将相同StreamID的数据批量上报。发送Demo可参考
->
SDK 内部会根据 StreamID 将相同 StreamID 的数据批量上报。发送 Demo 可参考

File: i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+整体流程包括以下三个步骤：
+
+### 初始化 SDK
+```go
+client, err := dataproxy.NewClient(
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md:27]: Please add a description for the init parameters.

File: i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
+---
+title: Go SDK
+sidebar_position: 4
+---
+
+import {siteVariables} from '../../version';
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md:6]: Remove this line, it is used for the defined version.

File: docs/sdk/dataproxy-sdk/go.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+	dataproxy.WithMetricsName("test"),
+)
+
+if err != nil {
+	fmt.Println(err)
+	return
+}
+```
+
+where:
```
Comment (dockerzhang) [inline docs/sdk/dataproxy-sdk/go.md:35]: Parameters:

File: docs/sdk/dataproxy-sdk/go.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
+- `dataproxy.WithGroupID("test")` sets the GroupID;
+- `dataproxy.WithURL("http://127.0.0.1:8083/inlong/manager/openapi/dataproxy/getIpList")` sets the service registry URL;
```
Comment (dockerzhang) [inline docs/sdk/dataproxy-sdk/go.md:38]: sets the service registry URL
->
sets the Manager URL

File: i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+if err != nil {
+	fmt.Println(err)
+	return
+}
+```
+
+参数说明：
+
+- `dataproxy.WithGroupID("test")` 设置了 GroupID ；
+- `dataproxy.WithURL("http://127.0.0.1:8083/inlong/manager/openapi/dataproxy/getIpList")` 设置了服务注册中心的 URL ；
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/version-1.7.0/sdk/dataproxy-sdk/go.md:38]: 设置了服务注册中心的 URL
->
设置 Manager URL

**Review by dockerzhang — APPROVED** [review]

**Review by gosonzhang — APPROVED** [review]

**Pull request merged**